### PR TITLE
feat: adds CR to variants per env UI

### PR DESCRIPTION
**Pull request by nunogois**
https://linear.app/unleash/issue/2-585/add-cr-to-variants-per-environment-ui

Adds CR to the variants per environment UI. This is basically the point where we have CRs integrated but can e.g. only update the weight once per CR. Adapting the UI to better fit CR logic will come next.

image

### CODE REVIEW

File: frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/EnvironmentVariantModal/EnvironmentVariantModal.tsx [hunk 1]
```diff
@@ -349,6 +387,29 @@ export const EnvironmentVariantModal = ({
...
+                                <ConditionallyRender
+                                    condition={Boolean(isChangeRequest)}
+                                    show={
+                                        <StyledCRAlert severity="info">
+                                            <strong>Change requests</strong> are
+                                            enabled
+                                            {environment
+                                                ? ` for ${environment.name}`
+                                                : ''}
+                                            . Your changes needs to be approved
```
Comment (sighphyre) [inline frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/EnvironmentVariantModal/EnvironmentVariantModal.tsx:403]:
```suggestion
                                            . Your changes need to be approved
```

**Review by sighphyre — APPROVED** [review]
Lots of branches in this. It _looks_ okay to me but it's hard to assess

File: frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/EnvironmentVariantModal/EnvironmentVariantModal.tsx (hunk 1 above)
Comment (nunogois) [reply frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/EnvironmentVariantModal/EnvironmentVariantModal.tsx]: Nice catch! https://github.com/Unleash/unleash/pull/2989/commits/d9aaaab27224beb416ffaa578bc6d2e5b168a0f2

File: frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/FeatureEnvironmentVariants.tsx [hunk 2]
```diff
@@ -46,14 +54,29 @@ export const FeatureEnvironmentVariants = () => {
...
+    const environments: IFeatureEnvironmentWithCrEnabled[] = useMemo(
+        () =>
+            feature?.environments?.map(environment => ({
+                ...environment,
+                crEnabled:
+                    uiConfig.flags.crOnVariants &&
+                    isChangeRequestConfigured(environment.name),
+            })) || [],
+        [feature.environments, uiConfig.flags.crOnVariants]
+    );
```
Comment (gastonfournier) [inline frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/FeatureEnvironmentVariants.tsx:78]: Oh, this is where the magic happens decorating the feature with `crEnabled` and this basically includes the `isChangeRequestConfigured` in env and feature toggle is enabled

Comment (nunogois) [reply frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/FeatureEnvironmentVariants.tsx:78]: Yeah exactly. This way we can clean this up once we make the flag GA and the logic is the same. Does this answer https://github.com/Unleash/unleash/pull/2989#discussion_r1086617295?

**Review by gastonfournier — APPROVED** [review]
Looks fine! The behavior of pushing variants when CR is not enabled in some env but pushing the CR in the case it's enabled looks neat, but might be confusing. Something we'll figure out once launched.

File: frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/FeatureEnvironmentVariants.tsx (hunk 2 above)
Comment (gastonfournier) [reply frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/FeatureEnvironmentVariants.tsx:78]: yes

**Pull request merged**